examples/s3/bucket-with-policy.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: s3.aws.crossplane.io/v1beta1
+kind: Bucket
+metadata:
+  name: test-bucket-with-policy
+  annotations:
+    # This will be the actual bucket name. It must be globally unique, so you
+    # probably want to change it before trying to apply this example.
+    crossplane.io/external-name: crossplane-example-bucket
+spec:
+  forProvider:
+    acl: private
+    locationConstraint: us-east-1
+    policy:
+      version: '2012-10-17'
+      statements:
+        - action:
+            - s3:ListBucket
+            - s3:GetBucketLocation
+            - s3:ListBucketMultipartUploads
+            - s3:PutBucketCORS
+          effect: Allow
+          principal:
+            awsPrincipals:
+              - iamUserArnSelector:
+                  matchLabels:
+                    example: "true"
+          resource:
```

Comment:
is self-referencing here not an option ?

```
  resourceRef:
    name: test-bucket-with-policy
    policy:
      resolution: Optional
      resolve: Always
```

think since https://github.com/crossplane-contrib/provider-aws/pull/1361 self-referencing is possible

Reply:
The policy schema is just copied from `BucketPolicy`. It does not currently support referencers.

Reply:
but we can add reference generation annotations and then it is working ? 
https://github.com/crossplane-contrib/provider-aws/pull/1685/files#diff-be80b55e7e7d430920dc7fc28366d7e7daec24b3ac55cbc127c3480b2f177e12R78

Reply:
I don't think a referencer is necessary here since the resource can only be ARN of the bucket we are managing. I added a step in the controller that late initializes these fields if not set.